js/state/siteUtil.js

```diff
@@ -444,14 +444,69 @@ module.exports.getDetailFromFrame = function (frame, tag) {
   })
 }
 
-module.exports.getDetailFromTab = function (tab, tag) {
+const getSitesLikeKey = (sites, siteKey, tag) => {
+  if (!sites || !siteKey) {
+    return makeImmutable([])
+  }
+  const splitKey = siteKey.split('|', 2)
+  const partialKey = splitKey.join('|')
+  const matches = sites.filter((site, key) => {
+    if (key.indexOf(partialKey) > -1 && (!tag || (tag && site.get('tags').includes(tag)))) {
+      return true
+    }
+    return false
+  })
+  return matches.toList()
+}
+
+module.exports.getDetailFromTab = function (tab, tag, sites) {
+  let location = tab.get('url')
+  const partitionNumber = tab.get('partitionNumber') !== undefined
+    ? tab.get('partitionNumber')
+    : undefined
```

Comment:
I left as undefined so that it would skip the below code, which only sets if the value is truthy:
```
if (partitionNumber) {
  siteDetail.partitionNumber = partitionNumber
```